src/xdg.c

```diff
@@ -418,8 +418,14 @@ handle_request_show_window_menu(struct wl_listener *listener, void *data)
 	struct xdg_toplevel_view *xdg_toplevel_view = wl_container_of(
 		listener, xdg_toplevel_view, request_show_window_menu);
 	struct server *server = xdg_toplevel_view->base.server;
+
 	struct menu *menu = menu_get_by_id(server, "client-menu");
 	assert(menu);
+
+	/* do the same as show_menu() in action.c */
+	menu->triggered_by_view = &xdg_toplevel_view->base;
+	server->menu_current = menu;
```

Comment:
@johanmalm @droc12345
Is `server->menu_current = menu;` needed?

It's added in #2102 and I blindly copied it from `show_menu()`, but looks like it's handled in `menu_open_root()`.

Reply:
Have had a look at the code. Agree, I don't think it's needed as set in `menu_open_root()`. 

I think we can remove that line from `show_menu()` too.